src/main/java/com/zaxxer/hikari/HikariJNDIFactory.java

```diff
@@ -40,31 +40,26 @@
 public class HikariJNDIFactory implements ObjectFactory
 {
    @Override
-   synchronized public Object getObjectInstance(Object obj, Name name, Context nameCtx, Hashtable<?, ?> environment) throws Exception
-   {
+   synchronized public Object getObjectInstance(Object obj, Name name, Context nameCtx, Hashtable<?, ?> environment) throws Exception {
       // We only know how to deal with <code>javax.naming.Reference</code> that specify a class name of "javax.sql.DataSource"
-      if (!(obj instanceof Reference)) {
-         return null;
-      }
-
-      Reference ref = (Reference) obj;
-      if (!"javax.sql.DataSource".equals(ref.getClassName())) {
-         throw new NamingException(ref.getClassName() + " is not a valid class name/type for this JNDI factory.");
-      }
+      if (obj instanceof Reference && "javax.sql.DataSource".equals(((Reference) obj).getClassName())) {
+         Reference ref = (Reference) obj;
+         Set<String> hikariPropSet = PropertyElf.getPropertyNames(HikariConfig.class);
 
-      Set<String> hikariPropSet = PropertyElf.getPropertyNames(HikariConfig.class);
-
-      Properties properties = new Properties();
-      Enumeration<RefAddr> enumeration = ref.getAll();
-      while (enumeration.hasMoreElements()) {
-         RefAddr element = enumeration.nextElement();
-         String type = element.getType();
-         if (type.startsWith("dataSource.") || hikariPropSet.contains(type)) {
-            properties.setProperty(type, element.getContent().toString());
+         Properties properties = new Properties();
+         Enumeration<RefAddr> enumeration = ref.getAll();
+         while (enumeration.hasMoreElements()) {
+            RefAddr element = enumeration.nextElement();
+            String type = element.getType();
+            if (type.startsWith("dataSource.") || hikariPropSet.contains(type)) {
+               properties.setProperty(type, element.getContent().toString());
+            }
          }
+         return createDataSource(properties, nameCtx);
+      }
+      else {
+         return null;
       }
```

Comment:
Eliminate the ``else`` and simply return null after the ``if``.

Reply:
DONE I am not a friend of multiple returns in a method because one have to read through all the method body to finally find out that some statements are never reached. The if-else block would have made it a little bit clearer.